backend/api/EventHandlers/MqttEventHandler.cs

```diff
@@ -352,9 +352,9 @@ private async void OnMissionUpdate(object? sender, MqttReceivedArgs mqttArgs)
                 if (flotillaMissionRun.MissionId != null)
                 {
                     var missionDefinition = await missionDefinitionService.ReadById(flotillaMissionRun.MissionId);
-                    if (missionDefinition != null)
+                    if (missionDefinition != null && flotillaMissionRun.Status == MissionStatus.Successful)
                     {
-                        missionDefinition.LastRun = flotillaMissionRun;
+                        missionDefinition.LastSuccessfulRun = flotillaMissionRun;
```

Comment:
Should this be moved into a scope service instead?

Reply:
Based on our recent discussions regarding the problem of switching between different contexts, I'm not sure this needs to be changed.